devbox.go

```diff
@@ -104,9 +105,20 @@ func (d *Devbox) Plan() *plansdk.Plan {
 	basePlan := &plansdk.Plan{
 		DevPackages:     d.cfg.Packages,
 		RuntimePackages: d.cfg.Packages,
-		SharedPlan:      d.cfg.SharedPlan,
 	}
-	return plansdk.MergePlans(basePlan, planner.GetPlan(d.srcDir))
+	plannerPlans := planner.GetPlan(d.srcDir)
```

Comment:
I would say:
1. Change `planner.GetPlan` to return the plan from only a single planner (for now the first one that matches) instead of a merged plan.
2. Call `MergePlans` so that the fields from devbox.json override, and slices append. If the function doesn't behave like we want to, you can add a sister function `OverridePlan` or something like that. No need to call `mergo` in here.
3. If you could add a test for this logic, I'd really appreciate it!

Reply:
@loreto I'm not sure about 1, but I will do 2 and 3 in this PR!

Changing 1 will affect the `Shell` command as well, which I don't think we want to only return the packages from one valid plan. I think we might have to think a bit more on that.

Reply:
Aren't we trying to guarantee the shell behaves the same way as build and anything else, though?

Reply:
Are we allowing a planner that only supports shell for the time being and not build?

Reply:
@loreto I'd be happy to do 1 as a follow-up. Want to think about it more though. Thoughts?